src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs

```diff
@@ -190,6 +190,55 @@ public void SdkLookup_Global_Json_Patch_Rollup()
         }
 
         [Fact]
+        public void SdkLookup_Negative_Version()
+        {
+            var fixture = PreviouslyBuiltAndRestoredPortableTestProjectFixture
+                .Copy();
+
+            var dotnet = fixture.BuiltDotnet;
+
+            // Add a negative CLI version
+            AddAvailableSdkVersions(_exeSdkBaseDir, "-1.-1.-1");
+
+            // CWD: empty
+            // User: empty
+            // Exe: -1.-1.-1
+            // Expected: no compatible version and a specific error message
+            dotnet.Exec("help")
+                .WorkingDirectory(_currentWorkingDir)
+                .WithUserProfile(_userDir)
+                .Environment(s_DefaultEnvironment)
+                .EnvironmentVariable("DOTNET_MULTILEVEL_LOOKUP", "0")
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should()
+                .Fail()
+                .And
+                .HaveStdErrContaining("It was not possible to find any SDK version");
+
+            // Add specified CLI version
+            AddAvailableSdkVersions(_exeSdkBaseDir, "9999.0.4");
+
+            // CWD: empty
+            // User: empty
+            // Exe: -1.-1.-1, 9999.0.4
+            // Expected: 9999.0.4 from exe dir
+            dotnet.Exec("help")
+                .WorkingDirectory(_currentWorkingDir)
+                .WithUserProfile(_userDir)
+                .Environment(s_DefaultEnvironment)
+                .EnvironmentVariable("DOTNET_MULTILEVEL_LOOKUP", "0")
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdErrContaining(Path.Combine(_exeSelectedMessage, "9999.0.4", _dotnetSdkDllMessageTerminator));
+        }
```

Comment:
Tests running simultaneously could get the same ID (numbered directory) in a race condition. Not sure if that is the reason for the random failures I get.